js/modules/idle_detector.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-env browser */
+
+const EventEmitter = require('events');
+
+
+const POLL_INTERVAL_MS = 10 * 1000;
+const IDLE_THRESHOLD_MS = 25;
+
+class IdleDetector extends EventEmitter {
+  constructor() {
+    super();
+    this.handle = null;
+    this.timeoutId = null;
+  }
+
+  start() {
+    this._scheduleNextCallback();
+  }
+
+  stop() {
+    if (this.handle) {
+      cancelIdleCallback(this.handle);
+    }
+
+    if (this.timeoutId) {
+      clearTimeout(this.timeoutId);
+    }
+  }
+
+  _scheduleNextCallback() {
+    this.stop();
+    this.handle = requestIdleCallback((deadline) => {
```

Comment:
Might be nice to put a `window.` here to make it clear where this function is coming from.